src/public/js/rs.js

```diff
@@ -21,6 +105,94 @@ var RS = (function() {
     }, '');
   }
 
+  // Encodes using systematic BCH Reed-Solomon on Galois Field 256
+  // Message should be passed in as length k block
+  // k specifies message length
+  // n specifies block length (number of symbols transmitted)
+  // n - k is the number of code symbols added.
+  // Encoding procedure has the largest order coefficient on the left
+  // and the smallest on the right. IE: [1,2,3] -> x^2 + 2x + 3
+  function encodeRSBlock(msg, n, k) {
+    remainder = polynomialDiv(genPoly, msg, n, k)
+    if (remainder == null) {
+      return null;
+    }
+    codedMsg = new Uint8Array(n)
+    for (i = 0; i < k; i++) {
+      codedMsg[i] = msg[i];
+    }
+    for (i = k; i < n; i++) {
+      codedMsg[i] = remainder[i - k]
+    }
+  }
+  function decodeRSBlock(codedMsg, n, k) {
+    remainder = polynomialDiv(genPoly, codedMsg, n, k)
+    if (remainder == null) {
+      return null;
+    }
+    errors = false;
+    for (i = 0; i < n-k; i++) {
+      if (remainder[i] != 0) {
+        errors = true
+      }
+    }
```

Comment:
a toolchain library called underscorejs provides some nice functional tools which I like for things like this! with underscore, I'd use the [some function](http://underscorejs.org/#some) to write lines 133-138 like this:

```
var errors = _.some(remainder, function(coefficient) {
    return coefficient != 0;
});
```
(assuming I'm understanding correctly that `remainder.length` is `n-k`). This would be a little more efficient because `some` stops as soon as it finds an element which doesn't pass the predicate, but the loop runs through all of them no matter what.

If we want to use underscore, we should include `<script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>` right after the jquery script tag in [src/public/index.html](https://github.com/sarahwalters/rs-chat/blob/master/src/public/index.html). might also be able to use it to make some of the other looping code "prettier", if we want :) (though the loops are just fine, freely admit I'm super biased towards functional patterns)